### DISCUSSION TITLE
1 Billion Rows or From 1 to 1 Billion?

### DISCUSSION THREAD

**Discussion by jhonesto** [gunnarmorling/1brc#597] [Q&A] [answered]
The challenge is with 1 billion records? or from 1 to 1 billion? 

When I run the test.sh script, it is running the program with files from 1 to 10k records. 

Far below the amount of records I was expecting. 

I was expecting a file with at least 6 characters per line * 1 billion. e.g.  { a;1.0'\n' } = 6bytes  * 1 Billion  = ~ 6GB 

All of my tests will fail in this case.

Tks.

**Comment by gunnarmorling** [comment, marked as answer]
An implementation shouldn't make any assumptions on the number of lines. 1B
is what implementations are evaluated against, but of course they should be
able to handle any number of fewer or more rows.
